x/staking/keeper/delegation.go

```diff
@@ -785,7 +785,7 @@ func (k Keeper) CompleteUnbonding(ctx sdk.Context, delAddr sdk.AccAddress, valAd
 
 	// loop through all the entries and try to complete unbonding mature entries
 	for i := 0; i < len(ubd.Entries); i++ {
```

Comment:
I think we should change the for loop bound computation. I think the evaluation might be compiler dependent and not part of the language spec. It's generally a bit spooky to delete while iterating so even if it's fine, it will be more readable. I'll make another commit

https://stackoverflow.com/questions/41327984/does-go-runtime-evaluate-the-for-loop-condition-every-iteration